src/shared/components/mutationTable/column/ASCNCopyNumberColumnFormatter.tsx

```diff
@@ -0,0 +1,255 @@
+import * as React from 'react';
+import DefaultTooltip from "public-lib/components/defaultTooltip/DefaultTooltip";
+import 'rc-tooltip/assets/bootstrap_white.css';
+import {MolecularProfile, Mutation, ClinicalData} from "shared/api/generated/CBioPortalAPI";
+import {default as TableCellStatusIndicator, TableCellStatus} from "public-lib/components/TableCellStatus";
+
+export default class ASCNCopyNumberColumnFormatter {
+
+    private static ascnCallTable:{[key:string]:string} = {
+        "no WGD,0,0":"Homdel",
+        "no WGD,1,0":"Hetloss",
+        "no WGD,2,0":"CNLOH",
+        "no WGD,3,0":"CNLOH & Gain",
+        "no WGD,4,0":"CNLOH & Gain",
+        "no WGD,5,0":"Amp (LOH)",
+        "no WGD,6,0":"Amp (LOH)",
+        "no WGD,1,1":"Diploid",
+        "no WGD,2,1":"Gain",
+        "no WGD,3,1":"Gain",
+        "no WGD,4,1":"Amp",
+        "no WGD,5,1":"Amp",
+        "no WGD,6,1":"Amp",
+        "no WGD,2,2":"Tetraploid",
+        "no WGD,3,2":"Amp",
+        "no WGD,4,2":"Amp",
+        "no WGD,5,2":"Amp",
+        "no WGD,6,2":"Amp",
+        "no WGD,3,3":"Amp (Balanced)",
+        "no WGD,4,3":"Amp",
+        "no WGD,5,3":"Amp",
+        "no WGD,6,3":"Amp",
+        "WGD,0,0":"Homdel",
+        "WGD,1,0":"Loss Before & After",
+        "WGD,2,0":"Loss Before",
+        "WGD,3,0":"CNLOH Before & Loss",
+        "WGD,4,0":"CNLOH Before",
+        "WGD,5,0":"CNLOH Before & Gain",
+        "WGD,6,0":"Amp (LOH)",
+        "WGD,1,1":"Double Loss After",
+        "WGD,2,1":"Loss After",
+        "WGD,3,1":"CNLOH After",
+        "WGD,4,1":"Loss & Gain",
+        "WGD,5,1":"Amp",
+        "WGD,6,1":"Amp",
+        "WGD,2,2":"Tetraploid",
+        "WGD,3,2":"Gain",
+        "WGD,4,2":"Amp",
+        "WGD,5,2":"Amp",
+        "WGD,6,2":"Amp",
+        "WGD,3,3":"Amp (Balanced)",
+        "WGD,4,3":"Amp",
+        "WGD,5,3":"Amp",
+        "WGD,6,3":"Amp"
+    };
+
+    private static hasRequiredAscnData(mutation:Mutation):boolean {
+        if (mutation.alleleSpecificCopyNumber === undefined || 
+            mutation.alleleSpecificCopyNumber.totalCopyNumber === undefined || 
+            mutation.alleleSpecificCopyNumber.minorCopyNumber === undefined ||
+            mutation.alleleSpecificCopyNumber.ascnIntegerCopyNumber == undefined) {
+            return false;
+        }
+        return true;
```

Comment:
no if/else needed it. Just return the result

Reply:
Well, to be more specific: 
```suggestion
        return !(
            mutation.alleleSpecificCopyNumber === undefined || 
            mutation.alleleSpecificCopyNumber.totalCopyNumber === undefined || 
            mutation.alleleSpecificCopyNumber.minorCopyNumber === undefined ||
            mutation.alleleSpecificCopyNumber.ascnIntegerCopyNumber == undefined
        );
```